### PR TITLE
Use `gelu_erf` as activation function for HGF hidden_act `gelu` and `gelu_python`

### PR DESCRIPTION
**Pull request by se-schmitt**
- adds the activation function `gelu_noapprox` which calculates the full gelu (pytorch default case, see [here](https://pytorch.org/docs/stable/generated/torch.nn.GELU.html)).

Context: I have implemented a thermodynamic model based on a ChemBERTa model [here](https://github.com/ClapeyronThermo/Clapeyron.jl/tree/master/lib/ClapeyronHANNA). To obtain the same results (within numerical accuracy) as the original model (which was implemented using pytorch, see [here](https://github.com/tspecht93/HANNA)), the "full" gelu activation function is needed (locally tested). So I would be happy to have this feature added in Transformers.jl, so that I can submit the corresponding changes to Clapeyron.jl.

### CODE REVIEW

**Comment by se-schmitt** [comment]
@chengchingwen have you had a chance to look at this?

**Comment by chengchingwen** [comment]
If the gelu without approx is the default in pytorch, we should also think about having that implemented in NNlib.jl. The question here is that there is no `"gelu_noapprox"` in huggingface/transformers so it wouldn't be loaded automatically unless you manually update the config.

So I would suggest:
1. open an issue/PR in NNlib.jl with the gelu implemented with `erf`
2. instead of adding `"gelu_noapprox"` to `ACT2FN`, directly replace the `gelu` for `[gelu, gelu_python]`.
3. It would be better to have a small test testing the `gelu` values and gradients

**Comment by se-schmitt** [comment]
@chengchingwen The [PR in NNlib](https://github.com/FluxML/NNlib.jl/pull/629)  was merged. Now the hidden_act keywords `gelu` and `gelu_python` use the new `gelu_erf` as proposed.

**Comment by chengchingwen** [comment]
1. need to update compat lower bound for NNlib with that commit
2. We still need to do a `gelu_erf_forward_backward` like https://github.com/chengchingwen/Transformers.jl/blob/12ed656d75aa1d8f29f44145c9fed1e76a39bebe/src/layers/base.jl#L164 for `gelu_erf` so that the computation can be fused.

**Comment by se-schmitt** [comment]
Ah, missed that, sorry! Also updated the compat entry.

File: src/layers/base.jl [hunk 1]
```diff
@@ -174,6 +174,8 @@ function gelu_forward_backward(x)
     return (forward, backward)
 end
 
+gelu_erf_forward_backward(x) = (gelu_erf(x), NNlib.deriv_gelu_erf(x))
```
Comment (chengchingwen) [inline src/layers/base.jl:177]:
```suggestion
using SpecialFunctions: erf

function gelu_erf_forward_backward(x)
    TWO = oftf(x,2)
    SQRT2 = sqrt(TWO)
    X_RSQRT2 = x / SQRT2
    ERF_SQRT1 = 1 + erf(X_RSQRT2)
￼   Φ = ERF_SQRT1 / TWO
    HALF_X = x / TWO
￼   forward = HALF_X * ERF_SQRT1
    backward = Φ + X_RSQRT2 * exp(-x * HALF_X) / sqrt(oftf(x,π))
    return (forward, backward)
end
```
And add a dependency for SpecialFunctions.jl
